Different behaviour of `expectNoEvents()` method between versions `0.8.0` and `0.9.0` when testing `MutableStateFlow`

It seems that version `0.9.0` changed behaviour of `expectNoEvents()` method.

```
@Test
fun booleanFlowTest() = runTest {
    val booleanFlow = MutableStateFlow(false)
    booleanFlow.test {
        assertEquals(false, awaitItem())
        booleanFlow.value = true
        expectNoEvents()
    }
}
```

`0.8.0` -> test fails with `app.cash.turbine.AssertionError: Expected no events but found Item(true)` (which is correct behaviour)
`0.9.0` -> test passes

What is even weirder, using deprecated method `runBlockingTest` when testing with version `0.9.0` results in correct behaviour.

--- a/src/commonMain/kotlin/app/cash/turbine/flow.kt
+++ b/src/commonMain/kotlin/app/cash/turbine/flow.kt
@@ -20,6 +20,7 @@ import kotlin.time.Duration.Companion.seconds
 import kotlinx.coroutines.CancellationException
 import kotlinx.coroutines.CoroutineScope
 import kotlinx.coroutines.CoroutineStart.UNDISPATCHED
+import kotlinx.coroutines.Dispatchers.Unconfined
 import kotlinx.coroutines.channels.Channel
 import kotlinx.coroutines.channels.Channel.Factory.UNLIMITED
 import kotlinx.coroutines.coroutineScope
@@ -110,7 +111,7 @@ public fun <T> Flow<T>.testIn(scope: CoroutineScope): ReceiveTurbine<T> {
 private fun <T> Flow<T>.collectTurbineIn(scope: CoroutineScope): Turbine<T> {
   lateinit var channel: Channel<T>
 
-  val job = scope.launch(start = UNDISPATCHED) {
+  val job = scope.launch(Unconfined, start = UNDISPATCHED) {
     channel = collectIntoChannel(this)
   }
 

--- a/src/commonTest/kotlin/app/cash/turbine/FlowTest.kt
+++ b/src/commonTest/kotlin/app/cash/turbine/FlowTest.kt
@@ -29,6 +29,7 @@ import kotlinx.coroutines.NonCancellable
 import kotlinx.coroutines.channels.Channel
 import kotlinx.coroutines.channels.Channel.Factory.RENDEZVOUS
 import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.flow.MutableStateFlow
 import kotlinx.coroutines.flow.emitAll
 import kotlinx.coroutines.flow.emptyFlow
 import kotlinx.coroutines.flow.flow
@@ -469,6 +470,19 @@ class FlowTest {
         onCompleteContinue.complete(Unit)
         awaitComplete()
       }
+  }
+
+  @Test fun valuesDoNotConflate() = runTest {
+    val flow = MutableStateFlow(0)
+    flow.test {
+      flow.value = 1
+      flow.value = 2
+      flow.value = 3
+      assertEquals(0, awaitItem())
+      assertEquals(1, awaitItem())
+      assertEquals(2, awaitItem())
+      assertEquals(3, awaitItem())
+    }
   }
 
   @Test fun assertNullValuesWithExpectMostRecentItem() = runTest {

